Cargo.toml

```diff
@@ -26,6 +25,10 @@ thiserror = "1.0.38"
 paste = "1.0.11"
 log = { version = "0.4.17", optional = true }
 env_logger = { version = "0.10.0", optional = true }
+serde = { version = "1.0.160", features = ["derive"] }
```

Comment:
That is a lot of packets. We should make it a goal to reduce the external crate footprint again over time.

Reply:
Indeed, I plan to eventually move most of this to the frontend tool